src/app/data-entry-multi-form/data-entry-multi-form.component.ts

```diff
@@ -66,39 +70,50 @@ export class DataEntryMultiFormComponent implements OnInit {
         }
       });
     }
+    this._dataEntryStepsDef = this.getFlatElements()
   }
 
   onReviewAndSubmit() {
     console.log('Review and Submit');
   }
 
-  getField(id, steps, isChild = false) {
-
+  getField(id, steps) {
     steps.forEach((step, index) => {
-      // this._currentStepIndex = index;
       if (step.id === id) {
-        if (!isChild) {
-          this._currentStepIndex = index;
-        }
-        this._currentChildStepIndex = isChild ? index : -1;
-
         this._currentStep = step;
-
       } else {
-
         if (step.steps?.length > 0) {
-          this.getField(id, step.steps, true)
+          this.getField(id, step.steps)
         }
-
       }
     });
-    console.log(this._currentStepIndex, 'current')
-    console.log(this._currentChildStepIndex, 'child')
   }
-  onPanelChange(id: string) {
-    // this._currentStepIndex = this.dataEntryForm.steps.findIndex(step => step.id === id);
 
-    this.getField(id, this.dataEntryForm.steps, false)
+  public getFlatElements() {
+    const results = this.dataEntryForm.steps;
+    const flat: DataEntrySteps[] = [];
+    const flatten = (array: any) => {
+      for (let i in array) {
+        const item = array[i];
+        flat.push({
+          id: item.id,
+          text: item.label
+        });
+        if (
+          item['steps'] &&
+          item['steps'].length
+        ) {
+          flatten(item['steps']);
+        }
+      }
+    };
+    flatten(results);
+    return flat;
+  }
+
+  onPanelChange(id: string) {
+    this._currentStepIndex = this._dataEntryStepsDef.findIndex(item => item.id === id);
+    this.getField(id, this.dataEntryForm.steps)
```

Comment:
is this getField needed anymore?

Reply:
Yes, it's needed to get the current field